File: backend/app/schemas/ticket.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


# -----------------------------
# Create Ticket
# -----------------------------
class TicketCreate(BaseModel):
    title: str
    description: str
    priority: str


# -----------------------------
# Update Ticket
# -----------------------------
class TicketUpdate(BaseModel):
    title: str
    description: str
    priority: str
    status: str
    assigned_to: int | None = None


# -----------------------------
# Assign Ticket
# -----------------------------
class TicketAssign(BaseModel):
    assigned_to: int


# -----------------------------
# Update Status
# -----------------------------
class TicketStatusUpdate(BaseModel):
    status: str


# -----------------------------
# Ticket Response
# -----------------------------
class TicketResponse(BaseModel):
    id: int
    title: str
    description: str
    priority: str
    status: str
    created_by: int
    assigned_to: str | None = None
    created_at: datetime | None = None

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/types/ticket.ts
export interface Ticket {
  id: number;

  title: string;
  description: string;

  category?: string;
  subcategory?: string;
  keywords?: string[];

  confidence?: number;

  priority: string;
  priority_reason?: string;

  status: string;

  created_by: number;

  assigned_to: number | null;
  assigned_name?: string;

  assignment_reason?: string;

  ai_processed?: boolean;

  created_at: string;
}